src/wallet/wallet_tx_builder.cpp

```diff
@@ -0,0 +1,697 @@
+// Copyright (c) 2022 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#include "wallet/wallet_tx_builder.h"
+
+using namespace libzcash;
+
+int GetAnchorHeight(const CChain& chain, uint32_t anchorConfirmations)
+{
+    int nextBlockHeight = chain.Height() + 1;
+    return nextBlockHeight - anchorConfirmations;
```

Comment:
What prevents this from being called when the chain is not synced (so that possibly `nextBlockHeight < anchorConfirmations`, or the result is before the Sapling or NU5 activation heights)?

Reply:
We use the result of this to determine if we are post-NU5, which then determines if we can use Orchard, and thus if we need the Orchard root at this height.